core/field.ts

```diff
@@ -194,7 +235,7 @@ export abstract class Field<T = any> implements IASTNodeLocationSvg,
    * this parameter supports.
    */
   constructor(
-      value: T|Sentinel, opt_validator?: FieldValidator<T>|null,
+      value: U|Sentinel, opt_validator?: FieldValidator<T>|null,
```

Comment:
Making it optional would make sense to me. I left it required because of the naming convention, as you mentioned.

Although `Field` can't be used directly since it's an abstract, base class, any subclass that has an optional value would have to call `super(Field.SKIP_SETUP)` or directly handle the case of when the `opt_value === undefined`  in the `super` call since super wouldn't accept the possibly undefined value.

This *would* have the effect of calling `doValueInvalid_` immediately on creating the field, though subclasses, like `FieldCheckbox`, are doing this anyway. (I think we also mentioned earlier that this was totally valid.)

Reply:
I went ahead and made the change to optional.